src/log.js

```diff
@@ -315,8 +315,17 @@ export class Log {
    *     not evaluate to true.
    * @param {string=} opt_message The assertion message
    * @param {...*} var_args Arguments substituted into %s in the message.
-   * @return {T} The value of shouldBeTrueish.
+   * @return {R} The value of shouldBeTrueish.
+   * @throws {!Error} When `value` is `null` or `undefined`.
    * @template T
+   * @template R :=
+   *     mapunion(T, (V) =>
+   *         cond(eq(V, 'null'),
+   *             none(),
+   *             cond(eq(V, 'undefined'),
+   *                 none(),
+   *                 V)))
+   *  =:
    * @closurePrimitive {asserts.matchesReturn}
```

Comment:
😵 what does it all mean? is there an annotated version?

Reply:
this seems to be a type transformer feature for Template types. 

i more or less followed the assertExists annotations found in:
https://github.com/google/closure-library/blob/master/closure/goog/asserts/asserts.js#L192-L213

Reply:
oh cool, thanks. it looks like magic

Reply:
haha i don't fully grok it either 😛

Reply:
Maybe add this link to the comments

Reply:
@cramforce done.